src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs

```diff
@@ -183,15 +183,12 @@ public static bool IsPathRooted(string path)
             return path.Length > 0 && path[0] == DirectorySeparatorChar;
         }
 
-        private static unsafe byte[] CreateCryptoRandomByteArray(int byteLength)
+        private static byte[] CreateCryptoRandomByteArray(int byteLength)
         {
             var arr = new byte[byteLength];
-            fixed (byte* buf = arr)
+            if (!Interop.Crypto.GetRandomBytes(arr, arr.Length))
```

Comment:
With the default provider, at least, `RAND_bytes()` only fails if the entropy pool never cleared the minimum threshold.  Once it has returned an answer you're good for the rest of the process (at least with 0.9.8).

`RAND_poll` is "keep doing things with randomness sources until the entropy pool exceeds the minimum threshold"; if it returns `0` that means OpenSSL couldn't figure out how randomness works for the system (compile-time behavior).

And since MSDN says that System.Security.Cryptography.RandomNumberGenerator is the cryptographically-secure RNG, we can't have it returning pseudo-bytes (like Random :smile:).